controllers/devworkspace/solver/endpoint_exposer.go

```diff
@@ -239,18 +240,24 @@ func (e *IngressExposer) getIngressForService(endpoint *EndpointInfo, endpointSt
 	return ingress
 }
 
-func routeAnnotations(machineName string, endpointName string) map[string]string {
-	return map[string]string{
-		defaults.ConfigAnnotationEndpointName:  endpointName,
-		defaults.ConfigAnnotationComponentName: machineName,
+func routeAnnotations(endpointAnnotations map[string]string, machineName string, endpointName string) map[string]string {
+	annos := map[string]string{}
+	for k, v := range endpointAnnotations {
+		annos[k] = v
 	}
+	annos[defaults.ConfigAnnotationEndpointName] = endpointName
+	annos[defaults.ConfigAnnotationComponentName] = machineName
+	return annos
 }
 
-func finalizeIngressAnnotations(ingressAnnotations map[string]string, machineName string, endpointName string) map[string]string {
+func finalizeIngressAnnotations(ingressAnnotations, endpointAnnotations map[string]string, machineName string, endpointName string) map[string]string {
```

Comment:
Is the order of parameters ingressAnnotations and endpointAnnotations correct?

Reply:
Yes: `ingressAnnotations` refers to the annotations to add to the [ingress provided by the Che Cluster CR](https://github.com/eclipse-che/che-operator/blob/aa0150d74fe0db59f475192ba13230c68dfd7e72/api/v2/checluster_types.go#L244-L251), and `endpointAnnotatiosn` refers to the annotations provided by the devfile. It's worth ensuring we want the devfile annotations to be able to override the Che Cluster CR annotations, see https://github.com/eclipse-che/che-operator/pull/1881/files#r1714630847

When [we call this function](https://github.com/eclipse-che/che-operator/pull/1881/files#diff-d071c0e02e16bcc2c5c74251891232a82f99aa56ae415f7b4e12178e28bd7dccR201), we're first providing the IngressExposer's annotations (coming from the Che Cluster CR, see the last parameter of [this function call](https://github.com/eclipse-che/che-operator/blob/aa0150d74fe0db59f475192ba13230c68dfd7e72/controllers/devworkspace/solver/che_routing.go#L457)) and then the endpoint's annotations (coming from the devfile through the[ devworkspace routing's endpoint](https://github.com/eclipse-che/che-operator/pull/1881/files#diff-ebca2eefe12f7ba4a722c53d574ba1b2adee412909da8cdbc974c8f7fcbfb02fR517))
```go
Annotations:     finalizeIngressAnnotations(endpoint.annotations, e.ingressAnnotations, endpoint.componentName, endpoint.endpointName),
```

Hopefully this clarifies things - let me know if I misinterpreted your question.

Reply:
In line 201, should it be the other way around, like this?
```
Annotations:     finalizeIngressAnnotations(e.ingressAnnotations, endpoint.annotations, endpoint.componentName, endpoint.endpointName),
```

Reply:
Ah I see now, can't believe I made that mistake. Good catch @tolusha @dkwon17 I'll fix this immediately 👍

Reply:
Should be good now, see https://github.com/eclipse-che/che-operator/pull/1881/commits/d6c17245142d036418928b63ab4fb8ab4d8cd168.

Once verified and approved, I'll squash the fixup commit :)